README.md

```diff
@@ -1,15 +1,49 @@
-# Kubecost Helm chart
+# Kubecost Helm Chart
 
-This repository contains the Helm chart templates for the development of [Kubecost](https://www.kubecost.com/), an enterprise-grade application to monitor and manage Kubernetes spend. Please see the [website](https://www.kubecost.com/) for more details on what Kubecost can do for you and the official documentation [IBM Docs](https://www.ibm.com/docs/en/kubecost/self-hosted/2.x), or contact [team-kubecost@wwpdl.vnet.ibm.com](mailto:team-kubecost@wwpdl.vnet.ibm.com) for assistance.
+This repository contains the Helm chart templates for the development of [Kubecost](https://www.kubecost.com/), an enterprise-grade application to monitor and manage Kubernetes spend. Please see the [website](https://www.kubecost.com/) for more details on what Kubecost can do for you and the official documentation [IBM Docs](https://www.ibm.com/docs/en/kubecost/self-hosted/3.x), or contact [team-kubecost@wwpdl.vnet.ibm.com](mailto:team-kubecost@wwpdl.vnet.ibm.com) for assistance.
 
-## Version Support
+## 3.0 Technical Overview
 
 Kubecost strives to support as many versions of Kubernetes as possible. Below is the version support matrix which has been tested. Versions outside of the stated range may still work but are untested.
 
-| Chart Version                  | Kubernetes Min | Kubernetes Max |
-|--------------------------------|----------------|----------------|
-| 2.8                            | 1.22           | 1.34           |
-| 3.0                            | 1.29           | 1.34           |
+| Chart Version | Kubernetes Min | Kubernetes Max | Notes |
+|---------------|----------------|----------------|-------|
+| 2.8           | 1.22           | 1.34           | Final release of Kubecost 2.x       |
+| 2.9           | 1.22           | 1.34           | Intermediate step to upgrade to 3.x |
+| 3.0           | 1.29           | 1.34           | First release of Kubecost 3.x       |
+
+### Migration path from 2.x to 3.x
+
+In order to upgrade from Kubecost 2.x to 3.x, it is recommended that all agents be updated to 2.9 for two days prior to upgrading to 3.0. See <https://github.com/kubecost/kubecost/tree/v2.9/examples> for how to upgrade to 2.9 before upgrading to 3.0.
+
+### Key changes in 3.0
+
+| Version | Database | Metrics Source | Automated Savings |
+|---------|----------|----------------|------------------|
+| 2.x     | DuckDB | Prometheus | Primary cluster only |
+| 3.0     | ClickHouse | Direct  | All clusters |
+
+Due to the new database, a complete re-ingestion of data will begin as soon as 3.x is installed. This will take anywhere from 20 minutes to 2 days to complete depending on the size of the dataset and PVC performance. During this time, the UI will be available, but will show a progress indicator. Data will be imported from today and going backwards in time until the full history is available.
```

Comment:
```suggestion
Due to the new database, a complete re-ingestion of data will begin as soon as 3.x is installed. This will take anywhere from 20 minutes to 2 days to complete depending on the size of the dataset and performance of the storage backing the Persistent Volume. During this time, the UI will be available, but will show a progress indicator. Data will be imported from today and going backwards in time until the full history is available.
```